From 654c8876f93677915b1a009bc7f2421ab8750bf1 Mon Sep 17 00:00:00 2001
From: Marios Makassikis <mmakassikis@freebox.fr>
Date: Fri, 23 Jul 2021 12:58:41 +0900
Subject: [PATCH] ksmbd: Fix potential memory leak in tcp_destroy_socket()
Git-commit: 654c8876f93677915b1a009bc7f2421ab8750bf1
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


ksmbd_socket must be freed even if kernel_sock_shutdown() somehow fails.

Signed-off-by: Marios Makassikis <mmakassikis@freebox.fr>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/ksmbd/transport_tcp.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/fs/ksmbd/transport_tcp.c b/fs/ksmbd/transport_tcp.c
index 56ec11ff5a9f..dc15a5ecd2e0 100644
--- a/fs/ksmbd/transport_tcp.c
+++ b/fs/ksmbd/transport_tcp.c
@@ -381,8 +381,7 @@ static void tcp_destroy_socket(struct socket *ksmbd_socket)
 	ret = kernel_sock_shutdown(ksmbd_socket, SHUT_RDWR);
 	if (ret)
 		pr_err("Failed to shutdown socket: %d\n", ret);
-	else
-		sock_release(ksmbd_socket);
+	sock_release(ksmbd_socket);
 }
 
 /**
-- 
2.33.0

